components/playlist/browser/playlist_service.cc

```diff
@@ -177,11 +197,27 @@ bool PlaylistService::RemoveItemFromPlaylist(const PlaylistId& playlist_id,
                           ConvertPlaylistToValue(target_playlist));
   }
 
-  // TODO(sko) Once we can support to share an item between playlists, we should
-  // check if other playlists have this item before deleting this item
-  // permanantly.
-  if (remove_item)
-    DeletePlaylistItemData(*item_id);
+  GetPlaylistItem(
+      *item_id,
+      base::BindOnce(
+          [](PlaylistService* service, const std::string& playlist_id,
+             bool delete_item, mojom::PlaylistItemPtr item) {
+            if (delete_item && item->parent_lists.size() == 1) {
+              DCHECK_EQ(item->parent_lists.front(), playlist_id);
+              service->DeletePlaylistItemData(item->id);
+              return;
+            }
+
```

Comment:
Q) Do we have the situation that `delete_item` is `false` and item's parent_lists size is `1`?

Reply:
Yeah, when moving a item to another playlist, `delete_item` is false and parent list would be 0. So the item temporarily belongs to nothing. And then added to another list.